codegenerator/cli/templates/static/codegen/src/Benchmark.res

```diff
@@ -22,15 +22,53 @@ module MillisAccum = {
 }
 
 module SummaryData = {
+  module DataSet = {
+    type t = {
+      count: int,
+      min: float,
+      max: float,
+      sum: float,
+      sumOfSquares: float,
+      decimalPlaces: int,
+    }
+
+    let schema = S.schema(s => {
+      count: s.matches(S.int),
+      min: s.matches(S.float),
+      max: s.matches(S.float),
+      sum: s.matches(S.float),
+      sumOfSquares: s.matches(S.float),
+      decimalPlaces: s.matches(S.int),
+    })
+
+    let make = (val: float, ~decimalPlaces=2) => {
+      count: 1,
+      min: val,
+      max: val,
+      sum: val,
+      sumOfSquares: val *. val,
+      decimalPlaces,
+    }
+
+    let add = (self: t, val: float) => {
+      count: self.count + 1,
+      min: Pervasives.min(self.min, val),
+      max: Pervasives.max(self.max, val),
+      sum: self.sum +. val,
+      sumOfSquares: self.sumOfSquares +. val *. val,
```

Comment:
These are the two places where a long running benchmark could eventually overflow.

Reply:
This is no longer relevant I just changed this to using BigDecimal type.